prime-router/src/main/kotlin/azure/EmailEngineFunction.kt

```diff
@@ -136,16 +137,36 @@ class EmailScheduleEngine {
 
             // get the orgs to fire for
             orgs.forEach { org ->
+                val countOfRecords = workflowEngine.db.fetchDownloadableReportFiles(
+                    OffsetDateTime.now().minusDays(1L),
+                    org
+                ).size;     
+                val template = getTemplate(schedule.template, countOfRecords );     
                 val emails: Iterable<String> = if (schedule.emails.size > 0) schedule.emails else getEmails(org, logger)
-                logger.info("EmailEngineFunction:: processing organization $org within ${schedule.template}")
+                logger.info("EmailEngineFunction:: processing organization $org within ${template}")
                 emails.forEach { email ->
-                    logger.info("EmailEngineFunction:: sending email to $email")
-                    dispatchToSendGrid(schedule.template, listOf(email), logger)
+                    logger.info("EmailEngineFunction:: sending email template ${template} to $email")
+                    dispatchToSendGrid(template, listOf(email), logger)
                 }
             }
         }
     }
 
+    /**
+     * Get the template to use
```

Comment:
This would benefit from a description of what the template looks like to indicate why we are doing this split.